backend/app/api/v1/commons/telco.py

```diff
@@ -112,13 +130,21 @@ async def getFilterData(start_datetime: date, end_datetime: date, configpath: st
         "latest_time": "{}T23:59:59".format(end_datetime.strftime("%Y-%m-%d")),
         "output_mode": "json",
     }
-    searchList = " OR ".join(
+    test_type_filter = " OR ".join(
         ['test_type="{}"'.format(test_type) for test_type in test_types]
     )
+    searchList = test_type_filter
+    if filter:
+        filter_dict = utils.get_dict_from_qs(filter)
+        searchQuery = utils.construct_query(filter_dict)
+        if "benchmark" in filter_dict:
+            searchList = searchQuery
+        else:
+            searchList = searchQuery + " " + test_type_filter
+
```

Comment:
This seems to be repeated between `getFilterData` and `getData` ... can it be factored out into a helper function?